src/c_api/value.cpp

```diff
@@ -610,6 +611,23 @@ kuzu_state kuzu_value_get_timestamp_tz(kuzu_value* value, kuzu_timestamp_tz_t* o
     return KuzuSuccess;
 }
 
+kuzu_state kuzu_value_get_decimal_as_string(kuzu_value* value, char** out_result) {
+    auto v = static_cast<Value*>(value->_value);
+    auto logical_type_id = v->getDataType().getLogicalTypeID();
+    auto physical_type_id = v->getDataType().getPhysicalType();
+    if (logical_type_id != LogicalTypeID::DECIMAL) {
+        return KuzuError;
+    }
+    if (physical_type_id != PhysicalTypeID::INT16 && physical_type_id != PhysicalTypeID::INT32 && physical_type_id != PhysicalTypeID::INT64 && physical_type_id != PhysicalTypeID::INT128) {
+        return KuzuError;
+    }
```

Comment:
DO NOT check the physical id type. As I said, this has nothing to do with the API level.